linkerd/app/outbound/src/http/retry.rs

```diff
@@ -14,10 +13,11 @@ use linkerd_http_retry::{
     ReplayBody,
 };
 use linkerd_retry as retry;
-use std::sync::Arc;
+pub use retry::Budget;
+use std::{num::NonZeroU32, sync::Arc};
 
 pub fn layer<N>(
-    metrics: metrics::HttpProfileRouteRetry,
+    metrics: Option<metrics::HttpProfileRouteRetry>,
```

Comment:
I know that the existing retry metrics are a bit odd, but do we understand what visibility we are sacrificing by not porting them forward? What questions can we no longer answer for policy retries?